tools/cli/pkg/command/upgrade.go

```diff
@@ -57,11 +62,24 @@ func (cmd *UpgradeCommand) ValidateTransformUpgradeOpts() error {
 
 	// Validate schedule
 	if scheduleParam, ok := scheduleInputToParam[cmd.schedule]; ok {
-		cmd.orchestrationParams.Strategy.Schedule = scheduleParam
+		// TODO Remove schedule type maintenancewindow
+		if scheduleParam == scheduleInputToParam["maintenancewindow"] {
+			return fmt.Errorf("the schedule type '%s' is deprecated, please use the option '--maintenancewindow' instead", scheduleParam)
+		}
+		cmd.orchestrationParams.Strategy.Schedule = cmd.schedule
+	} else if _, err := time.Parse(time.RFC3339, cmd.schedule); err == nil {
+		cmd.orchestrationParams.Strategy.Schedule = cmd.schedule
 	} else {
 		return fmt.Errorf("invalid value for schedule: %s. Check kcp upgrade --help for more information", cmd.schedule)
 	}
 
+	// Validate schedule
+	if scheduleAfterParam, err := time.Parse("2006-12-31", string(cmd.scheduleAfter)); err != nil {
```

Comment:
I think we won't need to to attributes for parsing a schedule time. 
How about just the `orchestrationParams.Strategy.Schedule` is kept, and the `orchestrationParams.Strategy.ScheduleAfter` is dropped? And then for `orchestrationParams.Strategy.Schedule` we would support the following valid input:

- "now"
- "immediate"
- time.Parse("2006-12-31", ...
- time.Parse(time.RFC3339, ...

Also, in handler code and orchestration manager, we would just deal with this one attribute.